### ISSUE TITLE
<complex>: Remove complex<NonFloating>

### ISSUE THREAD

**Issue by AlexGuteniev** [microsoft/STL#1078]
`std::complex<NonFoating>` does not work as user may expect.

The Standard only specifies the behavior of `std::complex<T>` for cv-unqualified floating-point types `T`, so the implementation can have any behavior in this case, ranging from not compiling the instantiation at all to full support of non-standard floating-point types.

See #756 for more context.

An attempt was made to deprecate this extension.
It did not work due to Real World Code test suite failing on NVIDIA/cutlass#104
See #889

This enhancement is the complete removal of the primary template.
This is the ultimate version of #756 enhancement.
This is not acceptable currently, but may be acceptable for ABI-breaking release.

**vNext note**: Resolving this issue will require breaking binary compatibility. We won't be able to accept pull requests for this issue until the vNext branch is available. See #169 for more information.

**Comment by AlexGuteniev** [comment]
@StephanTLavavej , does it mean that vNext is allowed to purposely break some Real World Code?

**Comment by StephanTLavavej** [comment]
Yes; in addition to the bincompat breaks, source compat breaks are acceptable when they have a purpose and are worth the disruption they cause.

**Comment by AlexGuteniev** [comment]
Maybe delete with `static_assert` instead ofjust not defining

**Comment by StephanTLavavej** [comment]
I agree that a pure removal isn't an ABI break. Given that NVIDIA Cutlass was fixed 5 years ago, I think we could explore adding a `static_assert` with an escape hatch, waiting some number of release cycles, and then dropping the primary template.
